### PR TITLE
#5715 Fix id token fetch for exception reports

### LINKED ISSUE #5715: Error reports send unauthorized requests from content script
When sending error report from content script, `idToken` is `undefined` which causes `401 Unauthorized` error on FES.

It happens because in [doSendErrorToSharedTenantFes](https://github.com/FlowCrypt/flowcrypt-browser/blob/cfe8276dac2189f3a8167e7db9e060114d63fbc0/extension/js/common/platform/catch.ts#L312) we try to parse `acctEmail` from current url, but there is no such url parameter in content script, need to fetch current `acctEmail` from storage.

### PR DESCRIPTION
**Pull request by sosnovsky**
This PR fixes id token fetch for sending error reports from content script.

\+ replaces `{ line: 0, col: 0 }` with `{ line: 1, col: 1}` for reports with unknown error location (as backend expects positive number, and throws error when line or col is 0)

close #5715

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests will be added later (https://github.com/FlowCrypt/flowcrypt-browser/issues/5718)

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [ ] addresses the issue it closes (if any)
- [ ] code is readable and understandable
- [ ] is accompanied with tests, or tests are not needed
- [ ] is free of vulnerabilities
- [ ] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: extension/js/common/platform/catch.ts [hunk 1]
```diff
@@ -311,8 +312,13 @@ export class Catch {
...
     try {
-      const uncheckedUrlParams = Url.parse(['acctEmail']);
-      const acctEmail = String(uncheckedUrlParams.acctEmail);
+      let acctEmail = Url.parse(['acctEmail']).acctEmail;
+      if (acctEmail) {
+        acctEmail = String(acctEmail);
+      } else {
+        const acctEmails = await GlobalStore.acctEmailsGet();
+        acctEmail = acctEmails?.length ? acctEmails[0] : undefined;
+      }
```
Comment (ioanmo226) [inline extension/js/common/platform/catch.ts:321]: How about write like below?

```ts
const { acctEmail: parsedEmail } = Url.parse(['acctEmail']);
let acctEmail = parsedEmail 
    ? String(parsedEmail) 
    : (await GlobalStore.acctEmailsGet())?.[0];

```

Comment (ioanmo226) [reply extension/js/common/platform/catch.ts:321]: I think that is why we added safe casting (?) before accessing first element?

> ?.[0]

Comment (sosnovsky) [reply extension/js/common/platform/catch.ts]: Yeah, you're right - I thought `?` checks only for `undefined` or `null` values (similar to Swift behavior), but it works for empty arrays too. I updated code according to your suggestion, thanks!

**Review by ioanmo226 — APPROVED** [review]
👍

**Pull request merged**